spec/output/generator_spec_rust/union.x/MyXDR.rs

```diff
@@ -180,17 +190,167 @@ where
 {
 }
 
+/// `DepthLimiter` is a trait designed for managing the depth of recursive operations.
+/// It provides a mechanism to limit recursion depth, and defines the behavior upon
+/// entering and leaving a recursion level.
+pub trait DepthLimiter {
+    /// The error type returned by methods when they fail due to exceeding the depth limit.
+    type DepthError;
+
+    /// Defines the behavior for entering a new recursion level.
+    /// A `DepthError` is returned if the new level exceeds the depth limit.
+    fn enter(&self) -> core::result::Result<(), Self::DepthError>;
```

Comment:
Yes you are right. 
The only reason I did RefCell was to avoid mut borrow, which would then allow me to use the `DepthGuard` pattern. But cloning the type solves that problem. 
Also to mention, the `DepthGuard` pattern did not work out in the host since the `leave()` method needs to be fallible due to `try_borrow_or_err` on the host, so it ended up not mattering anyway.